File: move-zeroes.rb
# @param {Integer[]} nums
# @return {Void} Do not return anything, modify nums in-place instead.
def move_zeroes(nums)
	size_before = nums.size
	nums.each do |num|
		if num == 0 then 
			nums.delete(num)
		end
	end
	for i in 0..size_before-nums.size-1
		nums.push(0)
	end
    return nums
end

File: move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        fast = slow = 0
        while fast < len(nums):
            if nums[fast] != 0:
                nums[slow] = nums[fast]
                slow += 1
            fast += 1
        nums[slow:] = [0] * (len(nums) - slow)
